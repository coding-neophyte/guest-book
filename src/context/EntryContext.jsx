import { useState, useContext, createContext } from 'react'

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
    const [entry, setEntry] = useState([]);

    return <EntryContext.Provider value={{entry, setEntry}}>
        {children}
    </EntryContext.Provider>
}

const useEntry = () =>{
    const context= useContext(EntryContext);

    if(context === undefined){
        throw new Error('invalid provider')
    }
    return context
}

export {EntryProvider, useEntry}
