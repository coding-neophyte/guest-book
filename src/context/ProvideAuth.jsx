import { createContext, useState } from "react";


const AuthCtx = createContext();

const AuthProvider = ({ children }) => {
    const [user, setUser] = useState()

    const login = (username, password) => {
      const loginVerified =  username === process.env.AUTH_USERNAME && password === process.env.AUTH_PASSWORD;

      if(loginVerified){
          setUser(username)
      }
      return loginVerified;
    }

    const logout = () => {
        setUser(null)
    }

    return (
        <AuthCtx.Provider value={{ user, login, logout }}>{children}</AuthCtx.Provider>
    );
}

export { AuthCtx, AuthProvider }
