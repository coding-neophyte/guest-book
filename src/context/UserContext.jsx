import { useState, useContext, createContext } from 'react'

const UserContext = createContext()

const UserProvider = ({ children }) => {
    const [user, setUser] = useState({})

    return <UserContext.Provider value={{user, setUser}}>
        {children}
    </UserContext.Provider>
}


const UseUser = () => {
    const context = useContext(context)

    if(context === undefined){
        throw new Error('invalid provider')
    }
    return context 
}

export { UserProvider, UseUser }
