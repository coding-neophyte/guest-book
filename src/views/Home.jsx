import React from 'react'
import GuestBook from '../components/GuestBook/GuestBook'
import EntryList from '../components/EntryList/EntryList'
import { EntryProvider } from '../context/EntryContext'

export default function Home() {
    return (
        <div>
            <EntryProvider>
            <GuestBook />
            <EntryList />
            </EntryProvider>
        </div>
    )
}
