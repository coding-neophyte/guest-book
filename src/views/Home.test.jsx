import React from 'react'
import { render } from '@testing-library/react'
import { EntryProvider } from '../context/EntryContext'
import { AuthProvider } from '../context/ProvideAuth'
import Home from './Home'

it('should render Home page', () => {
  const { container } = render(
    <EntryProvider>
    <AuthProvider>
        <Home />
     </AuthProvider>
    </EntryProvider>)
  expect(container).toMatchSnapshot()
})
