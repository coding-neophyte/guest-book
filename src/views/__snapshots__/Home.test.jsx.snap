// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home page 1`] = `
<div>
  <div>
    <div>
      <h2
        style="text-align: center;"
      >
         
        Please Sign the Guestbook
         
      </h2>
      <form>
        <label
          for="guestname"
        >
          Guest Name:
        </label>
        <input
          class="nameInput"
          id="guestname"
          name="guestname"
          type="text"
          value=""
        />
        <label
          for="guestentry"
        >
          Guest Entry
        </label>
        <textarea
          id="guestentry"
          name="guestentry"
          type="textarea"
        />
        <button
          type="submit"
        >
          Sign
        </button>
      </form>
    </div>
    <div>
      <ul />
    </div>
  </div>
</div>
`;
